tests/e2e_logging.py

```diff
@@ -794,6 +794,102 @@ def id_for(i):
     return network
 
 
+@reqs.description("Read state at multiple distinct historical points")
+@reqs.supports_methods("log/private", "log/private/historical/multi")
+def test_historical_query_multi(network, args):
+    idx = 142
+
+    seqnos = []
+
+    primary, _ = network.find_primary()
+    with primary.client("user0") as c:
+        # Submit many transactions, overwriting the same ID
+        # Need to submit through network.txs so these can be verified at shutdown, but also need to submit one at a
+        # time to retrieve the submitted transactions
+        msgs = {}
+        n_entries = 100
+
+        for _ in range(n_entries):
+            network.txs.issue(
+                network,
+                repeat=True,
+                idx=idx,
+                wait_for_sync=False,
+                log_capture=[],
+                send_public=False,
```

Comment:
I've added this `send_public=False` option because what I'd like to do here is submit a huge number of entries, and then check that fetching a few small sub-ranges is relatively fast, even if we do so in a single request. But if we interleave public and private entries (as we do by default, submitting one of each in each iteration), then we'll never get contiguous ranges of private seqnos to fetch.

Reply:
I continue to be skeptical of the value of the TxIssuer, and I worry that we're headed down the flag explosion route. It seems to me that it's simple enough to submit what we want, and check for commit of the last tx that it shouldn't need a wrapper.
If we namespace the keys with the test name, this ought to be absolutely fine.